src/main/java/com/teragrep/pth_06/planner/offset/DatasourceOffset.java

```diff
@@ -107,8 +107,8 @@ public LongOffset getArchiveOffset() {
     public KafkaOffset getKafkaOffset() {
         KafkaOffset kafkaOffset = serializedDatasourceOffset.kafkaOffset;
 
-        if (kafkaOffset == null || kafkaOffset.getOffsetMap() == null) {
-            return null;
+        if (kafkaOffset.isStub() || kafkaOffset.getOffsetMap().isEmpty()) {
```

Comment:
Not 100% sure about this, maybe it should throw an exception instead? Perhaps for another issue.

Reply:
In the original code it returns null. But after some research it seems like throwing exception here would be the most logical thing to do instead of returning anything.
Solved in commit ce4e405574cdec9aa904af88756e026f7c5bbede